### PR TITLE
Issue #3427929: Add Configuration Page from Social Follow User to Dashboard Configuration Menu

### PR DESCRIPTION
**Pull request by vcsvinicius**
## Problem
The configuration page to enable Follow User from Social Follow User module is in Administrator menu, but, ins't in Dashboard Configuration menu.

The site-managers don't have a menu to access this page, only by url.

## Solution
Add a menu for Configuration Follow User to Dashboard Configuration.
Menu should be: Dashboard -> Configuration -> Follow user settings

## Issue tracker
[PROD-28519](https://getopensocial.atlassian.net/browse/PROD-28510)
[#3427929](https://www.drupal.org/project/social/issues/3427929)

## Theme issue tracker
N/A

## How to test 
- [ ] Enable Social Follow User module
- [ ] Check menu: Dashboard -> Configuration -> Follow User Settings

### Before merge
- [ ] Code/peer review is completed
- [ ] All commit messages are [clear and clean](https://open-social.slite.com/app/docs/DnmermZDIx_0OQ). If applicable a rebase was performed
- [ ] All automated tests are green
- [ ] Functional/manual tests of the acceptance criteria are approved
- [ ] All acceptance criteria were met
- [ ] New features or changes to existing features are covered by tests, either unit (preferably) or behat
- [ ] Update path is tested. New hook_updates should respect update order, right naming convention and consider hook_post_update code
- [ ] Module can be safely uninstalled. Update/implement hook_uninstall and make sure that removed configuration or dependencies are removed/uninstalled
- [ ] This pull request has all required labels (team/type/priority)
- [ ] This pull request has a milestone
- [ ] This pull request has an assignee (if applicable)
- [ ] Any front end changes are tested on all major browsers
- [ ] New UI elements, or changes on UI elements are approved by the design team
- [ ] New features, or feature changes are approved by the product owner

### After merge
- [ ] Code is tested on all branches that it has been cherry-picked
- [ ] Update hook number might need adjustment, make sure they have the correct order
- [ ] The Drupal.org ticket(s) are updated according to this pull request status

## Screenshots
N/A

## Release notes
The Follow Uset Settings was added to Dashboard -> Configuration menu and it will be visible to site-manager user.

## Change Record
N/A

## Translations
N/A

[PROD-28519]: https://getopensocial.atlassian.net/browse/PROD-28519?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3817-nkljhx7cpshpud1ktikiccyxokzzlnac.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/65f33a23d95c0389470e93ef

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3817-nkljhx7cpshpud1ktikiccyxokzzlnac.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/65f33a23d95c0389470e93ef

File: modules/social_features/social_follow_user/social_follow_user.links.menu.yml [hunk 1]
```diff
@@ -3,3 +3,9 @@ social_follow_user.settings:
   description: 'Enable module functionality.'
   route_name: social_follow_user.settings
   parent: social_core.admin.config.social
+
+social_follow_user.social_core.dashboard.configuration.settings:
+  title: 'Follow user settings'
+  description: 'Enable module functionality.'
```
Comment (ronaldtebrake) [inline modules/social_features/social_follow_user/social_follow_user.links.menu.yml:9]:
```suggestion
  description: 'Enable functionality.'
```

Sounds more user friendly, module doesn't really say much to our SM 😊

**Comment by ronaldtebrake** [comment]
Works like a charm!

Perhaps we can also update 

```
  public function buildForm(array $form, FormStateInterface $form_state): array {
    $form['status'] = [
      '#type' => 'checkbox',
      '#title' => $this->t('Active'),
      '#default_value' => $this->config('social_follow_user.settings')->get('status'),
    ];
```

While we are at it, just to make sure it is a bit more user friendly.

Adding a description which states it activate / deactives the follow user functionality.

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3817-nkljhx7cpshpud1ktikiccyxokzzlnac.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/65f33a23d95c0389470e93ef

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3817-nkljhx7cpshpud1ktikiccyxokzzlnac.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/65f33a23d95c0389470e93ef

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3817-nkljhx7cpshpud1ktikiccyxokzzlnac.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/65f33a23d95c0389470e93ef

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3817-nkljhx7cpshpud1ktikiccyxokzzlnac.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/65f33a23d95c0389470e93ef

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3817-nkljhx7cpshpud1ktikiccyxokzzlnac.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/65f33a23d95c0389470e93ef

**Comment by BiaInacio** [comment]
Test done, menu item is displayed!
image

**Comment by vcsvinicius** [comment]
I cherry-picked to 12.4.x

**Pull request merged**